apps/studio/pages/api/ai/edge-function/complete.ts

```diff
@@ -0,0 +1,288 @@
+        3. For external imports, always define a version. For example, \`npm:@express\` should be written as \`npm:express@4.18.2\`.
+        4. For external dependencies, importing via \`npm:\` and \`jsr:\` is preferred. Minimize the use of imports from @\`deno.land/x\` , \`esm.sh\` and @\`unpkg.com\` . If you have a package from one of those CDNs, you can replace the CDN hostname with \`npm:\` specifier.
+        5. You can also use Node built-in APIs. You will need to import them using \`node:\` specifier. For example, to import Node process: \`import process from "node:process"\`. Use Node APIs when you find gaps in Deno APIs.
+        6. Do NOT use \`import { serve } from "https://deno.land/std@0.168.0/http/server.ts"\`. Instead use the built-in \`Deno.serve\`.
+        7. Following environment variables (ie. secrets) are pre-populated in both local and hosted Supabase environments. Users don't need to manually set them:
+          * SUPABASE_URL
+          * SUPABASE_ANON_KEY
+          * SUPABASE_SERVICE_ROLE_KEY
+          * SUPABASE_DB_URL
+        8. To set other environment variables the user can go to project settings then edge functions to set them
+        9. A single Edge Function can handle multiple routes. It is recommended to use a library like Express or Hono to handle the routes as it's easier for developer to understand and maintain. Each route must be prefixed with \`/function-name\` so they are routed correctly.
+        10. File write operations are ONLY permitted on \`/tmp\` directory. You can use either Deno or Node File APIs.
+        11. Use \`EdgeRuntime.waitUntil(promise)\` static method to run long-running tasks in the background without blocking response to a request. Do NOT assume it is available in the request / execution context.
+
+        ## Example Templates
+
+        ### Simple Hello World Function
+
+        \`\`\`edge
+        // Setup type definitions for built-in Supabase Runtime APIs
+        import "jsr:@supabase/functions-js/edge-runtime.d.ts";
+        interface reqPayload {
+          name: string;
+        }
+
+        console.info('server started');
+
+        Deno.serve(async (req: Request) => {
+          const { name }: reqPayload = await req.json();
+          const data = {
+            message: \`Hello \${name} from foo!\`,
+          };
+
+          return new Response(
+            JSON.stringify(data),
+            { headers: { 'Content-Type': 'application/json', 'Connection': 'keep-alive' }}
+          );
+        });
+        \`\`\`
+
+        ### Example Function using Node built-in API
+
+        \`\`\`edge
+        // Setup type definitions for built-in Supabase Runtime APIs
+        import "jsr:@supabase/functions-js/edge-runtime.d.ts";
+        import { randomBytes } from "node:crypto";
+        import { createServer } from "node:http";
+        import process from "node:process";
+
+        const generateRandomString = (length) => {
```

Comment:
all these too - can they be retrieved from Functions.templates.ts?

Reply:
Reluctant to change this as what is there is based off prompt example we give people. We might have a lot of templates that may add context duplication.